tests/utils.go

```diff
@@ -3962,35 +3946,29 @@ func StartTCPServer(vmi *v1.VirtualMachineInstance, port int) {
 	Expect(err).ToNot(HaveOccurred())
 }
 
+func StartHTTPServerWithLoggedInVMI(vmi *v1.VirtualMachineInstance, port int, isFedoraVM bool) {
+	var httpServerMaker string
+
+	if isFedoraVM {
+		httpServerMaker = fmt.Sprintf("python3 -m http.server %d --bind ::0 &", port)
+	} else {
+		httpServerMaker = fmt.Sprintf("screen -d -m nc -klp %d -e echo -e \"HTTP/1.1 200 OK\\nContent-Length: 11\\n\\nHello World!\"", port)
+	}
+	Expect(VmiConsoleRunCommand(vmi, httpServerMaker, 10*time.Second)).To(Succeed(), "should succeed starting http server")
+}
+
 func StartHTTPServer(vmi *v1.VirtualMachineInstance, port int, isFedoraVM bool) {
```

Comment:
It is clear now that the first part is about log-in and the 2nd part about starting the http server.
I guess @maiqueb PR fixes this, but if we just focus on this change, then just separating into two functions and calling each when needed is the correct path.

Reply:
I think one was missed. It relates also with the upper comment on the now helper functions which does not login.

Reply:
The networkpolicy tests is calling `StartHTTPServerWithLoggedInVMI` so I suppose we are good right ? or do you mean anything else ?

Reply:
I have refactored it closing this